src/App.tsx

```diff
@@ -16,60 +17,163 @@ const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  Alphabet,
+  Length,
+  None,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
+type State = {
+  goods: string[];
+  isVisible: boolean;
+  isReversed: boolean,
+  isSorted: boolean,
+  sortBy: SortType,
+};
+
+function getSorted(
   goods: string[],
-  sortType: SortType,
   isReversed: boolean,
+  sortBy: SortType,
+  isSorted: boolean,
 ) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  const copyGoods = [...goods];
+
+  if (isSorted) {
+    copyGoods.sort((good1, good2): number => {
+      switch (sortBy) {
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
 
-  // Sort and reverse goods if needed
-  // ...
+        default: return 0;
+      }
+    });
+  }
 
-  return visibleGoods;
+  if (isReversed) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
 }
 
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    isSorted: false,
+    sortBy: SortType.None,
+  };
+
+  visibleGoodsList = () => {
+    this.setState((state) => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+  reset = () => {
+    this.setState(() => ({
+      sortBy: SortType.None,
+      isReversed: false,
+      isSorted: false,
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState({
+      isSorted: true,
+      sortBy: SortType.Length,
+    });
+  };
+
+  sortByAlphabetic = () => {
+    this.setState({
+      isSorted: true,
+      sortBy: SortType.Alphabet,
+    });
+  };
```

Comment:
You can use one handler, just pass `sortBy` as parameter
```suggestion
  sortBy = (sortBy: SortType) => {
    this.setState({
      isSorted: true,
      sortBy,
    });
  };
```